auth/src/main/java/com/firebase/ui/auth/ui/email/CheckEmailFragment.java

```diff
@@ -100,9 +101,11 @@ public View onCreateView(LayoutInflater inflater,
         View v = inflater.inflate(R.layout.check_email_layout, container, false);
 
         // Email field and validator
+        mEmailLayout = (TextInputLayout) v.findViewById(R.id.email_layout);
         mEmailEditText = (EditText) v.findViewById(R.id.email);
-        mEmailFieldValidator = new EmailFieldValidator(
-                (TextInputLayout) v.findViewById(R.id.email_layout));
+        mEmailFieldValidator = new EmailFieldValidator(mEmailLayout);
+        mEmailLayout.setOnClickListener(this);
```

Comment:
@samtstern Right, I believe EmailFieldValidator doesn't have this behavior, the error message stays until the next button is clicked and validation is re-run.